scripts/build_osm_network.py

```
@@ -831,8 +831,12 @@ def built_network(inputs, outputs, geo_crs, distance_crs):
     country_list = input
     bus_country_list = buses["country"].unique().tolist()
 
-    if len(bus_country_list) != len(country_list):
-        no_data_countries = set(country_list).difference(set(bus_country_list))
+    # it may happen that bus_country_list contains entries not relevant as a country name (e.g. "not found")
+    # difference can't give negative values; the following will return only releant country names
+    no_data_countries = set(country_list).difference(set(bus_country_list))
+
+    if len(no_data_countries) > 0:
```

Very interesting work with sets!
Maybe we could do the check if set(country_list) != set(bus_country_list):
This may be more general and if that does not apply, then within the if we calculate the difference as you propose.
in such a case; it may be better to convert no_data_countries as a list; maybe something like `list(set(country_list).difference(set(bus_country_list)))`

We may use symmetric_difference instead of simple difference to generalize.
Most likely the difference would be negligible but by checking as proposed we miss whether some countries are included into bus_country_list and they are missing in country_list.
With simmetric_difference we keep this feature

Opinions?

---

Thank you :)

A very beautiful idea regarding `symmetric_difference()` and I'll definitely keep in mind this approach. But not sure if we may generalise this particular case. As far as I understand,  we have to treat a situation when there are some unexpected `country` entries in the buses dataframe in a different way as we do the countries without buses

Actually, if there are some unexpected countries in the buses dataframe, I'd rather raise an error as that is likely that something went wrong during OSM data cleaning (e.g. the data folders need some clean-up). Probably, it should be added to #528 

Do you agree?